Feature/view profile page

eugeneiv1 opened the pull request:
## Describe your changes

- add possibility to view public profile page of vendor/buyer
- add components of reviews and products
- add userApiSlice to fetch user data from server
- add follow feature

## Issue ticket code (and/or) and link

- [Link to JIRA ticket](https://olga-team.atlassian.net/browse/TP-181?atlOrigin=eyJpIjoiNWZlMjA3MmM4NTNiNGM5Njg2ZDE0Yzk5YWEwMGQ1MTMiLCJwIjoiaiJ9)

![followinf_buyer](https://github.com/ZenBit-Tech/faq_team_fe/assets/17147702/0077b29f-14ea-488d-b15a-1da6dbb6f008)
![Products](https://github.com/ZenBit-Tech/faq_team_fe/assets/17147702/bed1473b-c366-49b4-958f-4032dab10453)
![Reviews](https://github.com/ZenBit-Tech/faq_team_fe/assets/17147702/050e704e-320c-4033-b85f-7643d6ae170a)


### **General**

- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Types for input and output parameters
- [x] Don't have "any" on my code
- [ ] Used the try/catch pattern for error handling
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside the ternary operator
- [x] Don't have commented code
- [x] no links in the code, env links should be in env file (for
- [x] Used camelCase for variables and functions
- [x] Date and time formats are on the constants
- [x] Functions are public only if it's used outside the class
- [x] No hardcoded values
- [ ] covered by tests
- [x] Check your commit messages meet the [conventional commit f

### Frontend

- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

eugeneiv1 committed:
feat: add public profile sidebar and info components
feat: add public profile page

eugeneiv1 committed:
feat: add public profile sidebar and info components
feat: add public profile page
feat: add readmore component

eugeneiv1 committed:
feat: add public profile sidebar and info components
feat: add public profile page
feat: add readmore component

eugeneiv1 committed:
feat: add public profile sidebar and info components
feat: add public profile page
feat: add readmore component
feat: add carousel for product images
feat: add reviewList, reviewCard, productsList, productsCard components with styles

eugeneiv1 committed:
feat: add public profile sidebar and info components
feat: add public profile page
feat: add readmore component
feat: add carousel for product images
feat: add reviewList, reviewCard, productsList, productsCard components with styles

eugeneiv1 committed:
feat: add public profile sidebar and info components
feat: add public profile page
feat: add readmore component
feat: add carousel for product images
feat: add reviewList, reviewCard, productsList, productsCard components with styles
feat: add userApiSlice to get info about user from api

eugeneiv1 committed:
feat: add public profile sidebar and info components
feat: add public profile page
feat: add readmore component
feat: add carousel for product images
feat: add reviewList, reviewCard, productsList, productsCard components with styles
feat: add userApiSlice to get info about user from api

eugeneiv1 committed:
refactor: change rates prop to avgRate

MishaFomenko commented on src/components/productCard/index.tsx:
```diff
@@ -0,0 +1,32 @@
+} from 'components/productCard/styles.ts';
+import Carousel from 'components/carousel';
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
```
commented code

eugeneiv1 replied:
used to ignore unused variable until we fetch data from server

MishaFomenko commented on src/components/productCard/index.tsx:
```diff
@@ -0,0 +1,32 @@
+      <ProductName>Product Name</ProductName> {/*TODO change to fetched data*/}
+      <ProductInfo>
+        <div>
+          <ProductPrice>$ 213,99</ProductPrice>
```
price constant

eugeneiv1 replied:
the same TODO as for Product Name

MishaFomenko commented on src/components/publicProfileInfo/index.tsx:
```diff
@@ -0,0 +1,25 @@
+}: PublicProfileInfoType) => {
+  return (
+    <PublicProfileInfoWrapper>
+      {userRole === 'buyer' ? (
```
we should not compare with strings

eugeneiv1 replied:
fixed

MishaFomenko commented on src/components/publicProfileInfo/styles.ts:
```diff
@@ -0,0 +1,33 @@
+  .react-tabs__tab--selected {
+    width: 50%;
+    padding: 10px 0;
+    background-color: #333333;
```
const

eugeneiv1 replied:
fixed

MishaFomenko commented on src/components/publicProfileInfo/types.ts:
```diff
@@ -0,0 +1,7 @@
+import { ResponseGetProduct, UserReview } from '../../redux/types';
```
absolute path

eugeneiv1 replied:
fixed

MishaFomenko commented on src/components/publicProfileSidebar/index.tsx:
```diff
@@ -0,0 +1,77 @@
+
+  return (
+    <SideBarWrapper>
+      {/*TEXT IN ELEMENTS IS NOT FINAL RESULT!!! Just during feature in progress*/}
```
ok

MishaFomenko commented on src/components/publicProfileSidebar/index.tsx:
```diff
@@ -0,0 +1,77 @@
+      refetch();
+    } catch (err) {
+      if (isFetchBaseQueryError(err)) {
+        const errMsg = 'error' in err ? err.error : JSON.stringify(err.data);
```
comparing with strings

AlexandraVoloshyna replied:
This is not a comparison. There is nothing wrong here.

MishaFomenko commented on src/components/publicProfileSidebar/styles.ts:
```diff
@@ -0,0 +1,88 @@
+  margin-top: 32px;
+  padding: 14px 0;
+
+  border: 1px solid #edeae9;
```
color const

eugeneiv1 replied:
fixed

MishaFomenko commented on src/components/publicProfileSidebar/styles.ts:
```diff
@@ -0,0 +1,88 @@
+  font-family: ${({ theme }) => theme.fontNames.playfairDisplay};
+  text-align: center;
+  color: ${({ theme }) => theme.colors.white};
+  font-weight: 500;
```
const

eugeneiv1 replied:
fixed

AlexandraVoloshyna commented on src/components/publicProfileInfo/styles.ts:
```diff
@@ -0,0 +1,33 @@
+  .react-tabs__tab--selected {
+    width: 50%;
+    padding: 10px 0;
+    background-color: #333333;
```
background-color should be taken from the theme

eugeneiv1 replied:
fixed

eugeneiv1 committed:
fix: add const for user roles
fix: comparison to user role const in PublicProfileInfo
fix: relative path
fix: replace styles values to theme values

eugeneiv1 committed:
fix: add const for user roles
fix: comparison to user role const in PublicProfileInfo
fix: relative path
fix: replace styles values to theme values

eugeneiv1 committed:
Merge branch 'develop' into feature/view-profile-page

eugeneiv1 committed:
refactor: remove duplicates from package.json
